trie-db/src/lib.rs

```diff
@@ -376,30 +375,27 @@ impl<'db, L: TrieLayout> Trie<L> for TrieKinds<'db, L> {
 	}
 }
 
-impl<'db, L> TrieFactory<L>
-where
-	L: TrieLayout + 'db,
-{
+impl TrieFactory {
```

Comment:
`TrieFactory` is not used by substrate, but I feel like the change is good to have. Worth mentioning in CHANGELOG though.

Reply:
Here I just removed the warning of the unused `layout`

Reply:
I mean api did change (better `new`).